packages/autocertifier/src/CertificateCreator.ts

```diff
@@ -0,0 +1,111 @@
+import { Logger } from '@streamr/utils'
+import * as acme from 'acme-client'
+import fs from 'fs'
+import path from 'path'
+import { ChallengeManager } from './ChallengeManager'
+import { Certificate } from '@streamr/autocertifier-client'
+import os from 'os'
+
+const logger = new Logger(module)
+
+export class CertificateCreator {
+
+    private accountPrivateKey?: Buffer
+    private accountPrivateKeyPath: string
+    private readonly acmeDirectoryUrl: string
+    private readonly hmacKid: string
+    private readonly hmacKey: string
+    private readonly challengeManager: ChallengeManager
+
+    constructor(acmeDirectoryUrl: string,
+        hmacKid: string,
+        hmacKey: string,
+        privateKeyPath: string,
+        challengeManager: ChallengeManager
+    ) {
+        this.acmeDirectoryUrl = acmeDirectoryUrl
+        this.hmacKid = hmacKid
+        this.hmacKey = hmacKey
+        this.challengeManager = challengeManager
+
+        if (privateKeyPath.startsWith('~/')) {
+            this.accountPrivateKeyPath = privateKeyPath.replace('~', os.homedir())
+        } else {
+            this.accountPrivateKeyPath = privateKeyPath
+        }
+    }
+
+    public async createCertificate(fqdn: string): Promise<Certificate> {
+
+        logger.info(`Creating certificate for ${fqdn}`)
+        logger.info('Creating acme client')
+
+        const wasNewKeyCreated = await this.createPrivateKey()
+
+        const clientOptions: acme.ClientOptions = {
+            directoryUrl: this.acmeDirectoryUrl,
+            accountKey: this.accountPrivateKey!
+        }
+
+        if (wasNewKeyCreated) {
+            clientOptions.externalAccountBinding = {
+                kid: this.hmacKid,
+                hmacKey: this.hmacKey
+            }
+        }
+
+        const client = new acme.Client(clientOptions)
+
+        logger.info('Creating CSR')
+        const [key, csr] = await acme.crypto.createCsr({
+            commonName: fqdn
+        })
+
+        logger.info('Creating certificate using client.auto')
+        let cert: string
+
+        try {
+            cert = await client.auto({
+                csr,
+                email: 'autocertifier@streamr.network',
+                termsOfServiceAgreed: true,
+                challengePriority: ['dns-01'],
```

Comment:
Could introduce a constant for this (and link to https://letsencrypt.org/docs/challenge-types/ or other relevant URL)